### ISSUE TITLE
FNX-4959 ⁃ [Bug] `BrowserPerformanceTestActivity` no longer respects custom Intent arguments

### ISSUE THREAD

**Issue by ncalexan** [mozilla-mobile/fenix#1329]
## Steps to reproduce
```
adb shell am start -W -n org.mozilla.fenix.debug/org.mozilla.fenix.browser.BrowserPerformanceTestActivity --es args "-profile /mnt/sdcard/profile"
```

### Expected behavior
Custom profile is used.

### Actual behavior
Custom profile is not used.

Digging in, the Gecko runtime is being (lazily) created very early in the lifecycle, during [`FenixApplication.onCreate`](https://github.com/mozilla-mobile/fenix/blob/32478edf006ed374e48c0676910d9cdf3f2d61b4/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L75).  This happens before `BrowserPerformanceTestActivity.onCreate` has a chance to process its Intent arguments and set the `testConfig` for configuring the Gecko runtime.

My preferred way to address this issue is to [make GeckoView itself handle this configuration](https://bugzilla.mozilla.org/show_bug.cgi?id=1533385), and I'll try to work through some of these details right now.

In the interim, perhaps we could temporarily disable this metrics client work entirely for Raptor-enabled builds (i.e., those with `BrowserPerformanceTestActivity` at all) so that we unblock the Performance team.

**Comment by csadilek** [comment]
In the linked reference browser issue this was caused by the GeckoView fetch client passed to Glean (which was fixed/worked around by making the fetch client lazy).

In this case it's the GeckoView fetch client passed to service-fretboard / kinto which causes the same issue. The only quick workaround I see is to switch to `HttpUrlConnectionClient` until we have the fix in: https://bugzilla.mozilla.org/show_bug.cgi?id=1533385

I'll put a PR up.

**Comment by pocmo** [comment]
Is there a follow-up issue to get rid of `HttpURLConnectionClient`? That may be a release blocker.

**Comment by csadilek** [comment]
@pocmo I've filed one now: https://github.com/mozilla-mobile/fenix/issues/1350
